io_buffer.c

```diff
@@ -265,6 +265,28 @@ rb_io_buffer_type_allocate(VALUE self)
     return instance;
 }
 
+/*
+ *  call-seq: IO::Buffer.for(string) -> io_buffer
+ *
+ *  Creates read-write IO::Buffer from string's memory. The buffer remains associated
+ *  with the string, and writing to a buffer will reflect on string's content.
+ *
+ *  Note that as IO::Buffer is a low-level mechanism for efficient I/O operations, it
+ *  will ignore the string being frozen.
```

Comment:
That doesn't sound right semantically, it sounds more like a bug.
Also this could easily corrupt memory/make things crash/segfault if e.g. a fstring is passed and it's mutated by IO::Buffer.

Reply:
Yes you are right it was a bug, we have fixed it already. The buffer created from frozen string will be immutable.